packages/connectivity/src/scp-cf/destination/destination-from-service.ts

```diff
@@ -256,13 +256,17 @@ export class DestinationFromServiceRetriever {
     const { destination, origin } = destinationResult;
     // This covers the x-tenant case https://api.sap.com/api/SAP_CP_CF_Connectivity_Destination/resource
     const exchangeTenant = this.getExchangeTenant(destination);
-    const clientGrant = await serviceToken('destination', {
-      jwt:
-        origin === 'provider'
-          ? this.providerServiceToken.decoded
-          : this.subscriberToken?.serviceJwt?.decoded
-    });
-    return { authHeaderJwt: clientGrant, exchangeTenant };
+    const authHeaderJwt =
+      origin === 'provider'
+        ? this.providerServiceToken.encoded
+        : this.subscriberToken?.serviceJwt?.encoded;
+
+    if (!authHeaderJwt) {
```

Comment:
[q] Just wondering if this check here is needed again?

Reply:
I guess not. 

provider case: token is there anyways, so **no**
subscriber case: technically token could not be there, but if it is not there, the origin cannot be "subscriber" either, so I guess **no**

Reply:
But actually, after I checked, this is a type problem and I prefer to solve it like this instead of using `!`.